community/cypher/interpreted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/interpreted/pipes/NodeIndexSeeker.scala

```diff
@@ -52,7 +54,7 @@ trait NodeIndexSeeker {
       case _: ExactSeek |
            _: SeekByRange =>
         val indexQueries = computeIndexQueries(state, baseContext)
-        indexQueries.flatMap(query => state.query.indexSeek(indexReference, query)).toIterator
+        indexQueries.toIterator.flatMap(query => state.query.indexSeek(indexReference, query))
```

Comment:
Lazyness. Otherwise we create an eager `Seq[NodeValue]`, which we _then_ turn into an iterator. Like this we have a lazy `Iterator[NodeValue]` from the start